platform_plugin_forum_email_notifier/handlers.py

```diff
@@ -0,0 +1,52 @@
+"""Signal handlers for forum events."""
+from django.dispatch import receiver
+from openedx_events.learning.signals import FORUM_THREAD_CREATED, FORUM_THREAD_RESPONSE_CREATED, FORUM_RESPONSE_COMMENT_CREATED
+
+from platform_plugin_forum_email_notifier.tasks import send_email_notification
+from platform_plugin_forum_email_notifier.utils import get_subscribers
+
+
+@receiver(FORUM_THREAD_CREATED)
+def forum_thread_created_handler(signal, sender, thread, metadata, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handler for forum thread created event.
+    """
+    notify_users(thread, object_type="thread")
+
+
+@receiver(FORUM_THREAD_RESPONSE_CREATED)
+def forum_response_created_handler(signal, sender, thread, metadata, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handler for forum response created event.
+    """
+    notify_users(thread, object_type="response")
+
+
+@receiver(FORUM_RESPONSE_COMMENT_CREATED)
+def forum_comment_created_handler(signal, sender, thread, metadata, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handler for forum comment created event.
+    """
+    notify_users(thread, object_type="comment")
+
+
+def notify_users(thread, object_type):
```

Comment:
why isn't this a task as well? we could have a large number of subscribers

Reply:
Will be refactored in this PR https://github.com/eduNEXT/platform-plugin-forum-email-notifier/pull/3/files#diff-d6f8aa8c3a53ed681d96a188fd78330e8a56fafecdc0171b67c325cccb2648adR64-R101